lib/icinga/checkable-check.cpp

```diff
@@ -249,8 +249,13 @@ void Checkable::ProcessCheckResult(const CheckResult::Ptr& cr, const MessageOrig
 			if (parent.get() == this)
 				continue;
 
-			ObjectLock olock(parent);
-			parent->SetNextCheck(Utility::GetTime());
+			if (!parent->GetEnableActiveChecks())
+				continue;
+
+			if (parent->GetNextCheck() >= now + parent->GetRetryInterval()) {
```

Comment:
This doesn't feel right. Any explanation why one would use retry_interval here?

Reply:
Right now there is no limitation to the amount of reschedules that can be forced to a parent by a failed child check. In some cases with about 20 children for one NRPE parent, it will go through its max_check_attempts within a few seconds.

Conceptually the minimum waiting period between checks is the retry_interval during normal scheduling(thus excluding a manual forced reschedule). Theoretically it is possible to have a lower check_interval than retry_interval, however I doubt anyone does that as it makes little sense.

Reply:
Alright. I'd like to see this PR tested in various scenarios then, including test cases and excerpts on the next_check values from the API.